sources/models/src/lib.rs

```diff
@@ -198,6 +198,7 @@ struct KubernetesSettings {
     topology_manager_scope: TopologyManagerScope,
     topology_manager_policy: TopologyManagerPolicy,
     pod_pids_limit: i64,
+    provider_id: Url,
```

Comment:
Our `Url` type accepts strings with a scheme - is `kubelet` OK with this, or does it error out if given something that doesn't match `<ProviderName>://<ProviderSpecificNodeID>`?

Reply:
`kubelet` is ok with any string it seems. I tried both URL with scheme and without scheme and kubelet had no qualms with either and ran fine both times. It would be up to the operator/bootstrap process to set it correctly as expected by the CAPI controller.